pkg/infoschema/infoschema_v2.go

```diff
@@ -343,6 +343,15 @@ func (is *infoschemaV2) AllSchemas() (schemas []*model.DBInfo) {
 	return
 }
 
+func (is *infoschemaV2) AllSchemaNames() []model.CIStr {
+	rs := make([]model.CIStr, 0, is.Data.schemaMap.Len())
```

Comment:
You need to handle `is.Data.specials` like here https://github.com/pingcap/tidb/pull/51436/files#diff-d82e5bb6499a8d8c00f3d55ec95bccda3fab6d4cc45529cecf06552bfa973b1bL352-L355